internal/aws/k8s/k8sclient/obj_store.go

```diff
@@ -0,0 +1,156 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8sclient
+
+import (
+	"fmt"
+	"sync"
+
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type ObjStore struct {
+	sync.RWMutex
+
+	refreshed bool
+	objs      map[types.UID]interface{}
+
+	transformFunc func(interface{}) (interface{}, error)
+	logger        *zap.Logger
+}
+
+func NewObjStore(transformFunc func(interface{}) (interface{}, error), logger *zap.Logger) *ObjStore {
+	return &ObjStore{
+		transformFunc: transformFunc,
+		objs:          map[types.UID]interface{}{},
+		logger:        logger,
+	}
+}
+
+// Track whether the underlying data store is refreshed or not.
+// Calling this func itself will reset the state to false.
+func (s *ObjStore) GetResetRefreshStatus() bool {
+	s.Lock()
+	defer s.Unlock()
+
+	refreshed := s.refreshed
+	if refreshed {
+		s.refreshed = false
+	}
+	return refreshed
+}
+
+func (s *ObjStore) Add(obj interface{}) error {
+	o, err := meta.Accessor(obj)
```

Comment:
The main reason I ask about documentation on the k8s API is I'm wondering if this does an RPC to the k8s apiserer? If so, `ObjStore.Add` seems like a way too simple sounding method name to be doing an RPC